sdk/evm/tbrv3/contract.ts

```diff
@@ -112,364 +104,378 @@ export interface Transfer {
    * Increasing this estimation won't affect the actual cost.
    * Excedent gas tokens will be returned to the caller of the contract.
    */
-  feeEstimation: RelayingFee;
+  feeEstimation: RelayingFeeReturn;
   args: TransferTokenWithRelayInput | TransferGasTokenWithRelayInput;
 };
 
 export class Tbrv3 {
+  static readonly addresses = {
+    Mainnet: zeroAddress,
+    Testnet: zeroAddress,
+  } as const satisfies Record<NetworkMain, EvmAddress>;
 
-  readonly address: string;
+  /**
+   * Creates the initialization configuration for the TBRv3 proxy contract.
+   * 
+   * @param owner contract owner address, it must be a valid hex evm address
+   * @param admin admin address, it must be a valid hex evm address.
+   * @param feeRecipient fee recipient address, it must be a valid hex evm address.
+   * @returns The serialized layout of the initialization configuration.
+   */
+  static proxyConstructor(
+    owner: EvmAddress,
+    admins: EvmAddress[],
+    feeRecipient: EvmAddress,
+  ) {
+    const initConfig = {
+      owner,
+      admins,
+      feeRecipient,
+    };
 
-  constructor(public readonly provider: ethers.Provider, public readonly network: NetworkMain, address: string = Tbrv3.addresses[network]) {
-    this.address = address;
+    return serializeLayout(proxyConstructorLayout, initConfig);
   }
 
-  static readonly addresses: Record<NetworkMain, string> = {
-    Mainnet: "TBD",
-    Testnet: "TBD",
+  static connect(
+    provider: ConnectionPrimitives,
+    network: NetworkMain,
+    chain: Chain,
+    gasToken?: EvmAddress,
+    address?: EvmAddress,
+  ) {
+    if (address === undefined && this.addresses[network].equals(zeroAddress)) {
+      throw new Error(`Tbrv3 address needs to be provided for network ${network}`);
+    }
+
+    const defaultGasToken = getNative(network, chain);
+    if (gasToken === undefined) {
+      if (defaultGasToken === undefined) {
+        throw new Error(`Gas token address needs to be provided for network ${network} and chain ${chain}`);
+      }
+      gasToken = new EvmAddress(defaultGasToken.wrappedKey!);
```

Comment:
wrappedKey would be something like "WETH", so it seems we are missing an extra step like [the following](https://github.com/wormhole-foundation/wormhole-sdk-ts/blob/fa4ba4bc349a7caada809f209090d79a3c5962fe/core/definitions/src/types.ts#L109)?
```
const wrappedKey = defaultGasToken.wrappedKey!;
const wrappedToken = getTokenByKey(network, chain, wrappedKey);
if (!wrappedToken) throw new Error("Invalid wrapped token key: " + wrappedKey);
const destNativeWrapped = { chain, address: toNative(chain, wrappedToken.address) };
```

Reply:
Ah, yeah, probably :thinking:

Reply:
Let me check that there isn't a simpler way.

Reply:
We should probably just use that function instead, I think.

Reply:
Although in that case, we won't be throwing the errors ourselves.

Reply:
Fixed. A bit messy but it accomplishes the task :P